### ISSUE TITLE
Empty line separator is not enforced in `package-info.java` file

### ISSUE THREAD

**Issue by mohitsatr** [checkstyle/checkstyle#17541]
I have read check documentation: https://checkstyle.org/checks/whitespace/emptylineseparator.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

From: https://google.github.io/styleguide/javaguide.html#s3-source-file-structure
> Exactly one blank line separates each section that is present.
> A package-info.java file is the same, but without the top-level class.

$ cat package-info.java
```java
///////////////////////////////////////////////////////////////////////////////////////////////
// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
///////////////////////////////////////////////////////////////////////////////////////////////
package com.google.checkstyle.test.chapter3filestructure.rule3sourcefile;
// EXPECTED VIOLATION ABOVE

```

CLI:
```
$ java -jar checkstyle-10.26.1-all.jar -c google_checks.xml package-info.java
Starting audit...
Audit done.
```

Google-formatter also has issues with formatting:

```
$ java -jar google-java-format-1.27.0-all-deps.jar package-info.java > FormattedCode.java
$ diff -Naru package-info.java FormattedCode.java
```

**Comment by mohitsatr** [comment]
@romani @Zopsss 
The reason for missing coverage is because of design flaw in `EmptyLineSeparatorCheck`. It does not check line separation for the last token in the file.

https://github.com/checkstyle/checkstyle/blob/96badccf2ace21af141dd934e46427531261ca63/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/packageinfo/test3/package-info.java#L14-L19

**Comment by romani** [comment]
nice catch, we need to update Check to remove import from this Input  file.

Imports should be only if this is required for annotations
https://www.baeldung.com/java-package-info#package-annotations
```
@NonNullApi
@NonNullFields
package com.baeldung.nullibility;

import org.springframework.lang.NonNullApi;
import org.springframework.lang.NonNullFields;
```

**Comment by Zopsss** [comment]
@mohitsatr can you confirm is this only related to Package? or does it also affects other cases as well?

**Comment by romani** [comment]
if there is any other cases ... it should be separate ticket.

**Comment by romani** [comment]
Merged

**Comment by romani** [comment]
Thanks a lot to @MonuChaudhary14 for huge effort to make it done
